joerd/source/ned.py

```diff
@@ -0,0 +1,183 @@
+from joerd.util import BoundingBox
+from multiprocessing import Pool
+from contextlib import closing
+from shutil import copyfile
+from ftplib import FTP
+import os.path
+import os
+import requests
+import logging
+import re
+import tempfile
+import sys
+import zipfile
+import traceback
+import subprocess
+import glob
+from osgeo import gdal
+import urllib2
+import shutil
+
+
+NED_FTP_SERVER = 'rockyftp.cr.usgs.gov'
+NED_BASE_PATH = 'vdelivery/Datasets/Staged/NED/19/IMG'
+
+
+def __download_ned_file(img_name, zip_name, base_dir):
+    logger = logging.getLogger('ned')
+    output_file = os.path.join(base_dir, img_name)
+
+    if os.path.isfile(output_file):
+        return output_file
+
+    with closing(tempfile.NamedTemporaryFile()) as tmp:
+        url = 'ftp://%s/%s/%s' % (NED_FTP_SERVER, NED_BASE_PATH, zip_name)
+        logger.info("FTP: Fetching %r" % url)
+
+        with closing(urllib2.urlopen(url)) as req:
+            shutil.copyfileobj(req, tmp)
+
+        tmp.flush()
+
+        with zipfile.ZipFile(tmp.name, 'r') as zfile:
+            zfile.extract(img_name, base_dir)
+            zfile.extract(img_name + ".aux.xml", base_dir)
```

Comment:
Can we have the zip file take the stream directly? Or does it need the whole file in  advance?